libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLine.java

```diff
@@ -170,14 +177,28 @@
 
     // Waypoint attributes
     int originWaypointIcon = typedArray.getResourceId(
-      R.styleable.NavigationMapRoute_originWaypointIcon, R.drawable.ic_route_origin);
+            R.styleable.NavigationMapRoute_originWaypointIcon, R.drawable.ic_route_origin);
     int destinationWaypointIcon = typedArray.getResourceId(
-      R.styleable.NavigationMapRoute_destinationWaypointIcon, R.drawable.ic_route_destination);
+            R.styleable.NavigationMapRoute_destinationWaypointIcon, R.drawable.ic_route_destination);
     typedArray.recycle();
 
     Drawable originIcon = drawableProvider.retrieveDrawable(originWaypointIcon);
     Drawable destinationIcon = drawableProvider.retrieveDrawable(destinationWaypointIcon);
-    belowLayer = findRouteBelowLayerId(belowLayer, style);
+
+    boolean layerBelowExists = false;
+    if (belowLayer != null && !belowLayer.isEmpty()) {
+      for (Layer layer : style.getLayers()) {
+        if (layer.getId().equals(belowLayer)) {
+          layerBelowExists = true;
+          break;
+        }
+      }
+    }
+
+    if (!layerBelowExists) {
+      // Avoid placing the route on top of the user location layer
+      belowLayer = LocationComponentConstants.SHADOW_LAYER;
```

Comment:
We are regressing here ever so slightly - if there's no layer below defined by the user, we should find the indices of `LocationComponentConstants.SHADOW_LAYER` and `RouteConstants.LAYER_ABOVE_UPCOMING_MANEUVER_ARROW` in the layers list above and place the route layer below one of those layers which is lower in the stack.
The regression can be seen running the `NavigationMapRouteActivity` where the deprecated Maps markers are below the route line. Even though they are deprecated, we should try to avoid regressing.

Reply:
@LukasPaczos it's a great explanation, thank you! gonna implement it

Reply:
fixed, please check it